src/main/java/io/druid/data/input/impl/RegexParseSpec.java

```diff
@@ -39,13 +41,18 @@ public RegexParseSpec(
       @JsonProperty("timestampSpec") TimestampSpec timestampSpec,
       @JsonProperty("dimensionsSpec") DimensionsSpec dimensionsSpec,
       @JsonProperty("listDelimiter") String listDelimiter,
+      @JsonProperty("columns") List<String> columns,
       @JsonProperty("pattern") String pattern
   )
   {
     super(timestampSpec, dimensionsSpec);
 
     this.listDelimiter = listDelimiter;
+    Preconditions.checkNotNull(columns, "columns");
+    this.columns = columns;
     this.pattern = pattern;
+
+    verify(dimensionsSpec.getDimensions());
```

Comment:
what happens if user did not explicitly specify dimensions (schemaLess ingestion)?